### PR TITLE
refactor: sqlite with SQLAlchemy

### LINKED ISSUE #2743: [persistence] use sql alchemy

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #2743

### CODE REVIEW

File: src/phoenix/db/models.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            name="project_name_unique",
+            sqlite_on_conflict="IGNORE",
+        ),
+    )
+
+
+class Trace(Base):
+    __tablename__ = "traces"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    project_rowid: Mapped[int] = mapped_column(ForeignKey("projects.id"))
```
Comment (mikeldking) [inline src/phoenix/db/models.py:50]: Any idea if delete cascade can be setup on these?

Comment (RogerHYang) [reply src/phoenix/db/models.py:51]: I added (delete-orphan) cascade to the (one-to-many) relationship.

File: src/phoenix/db/models.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    session_id: Mapped[Optional[str]]
```
[the remark this answers is not in the document]
Comment (RogerHYang) [reply src/phoenix/db/models.py:52]: ok, thx

File: src/phoenix/db/database.py [hunk 3]
```diff
@@ -51,24 +62,43 @@
...
+
+@event.listens_for(Engine, "connect")
+def set_sqlite_pragma(dbapi_connection: Connection, _: Any) -> None:
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA foreign_keys = ON;")
+    cursor.execute("PRAGMA journal_mode = WAL;")
+    cursor.execute("PRAGMA synchronous = OFF;")
+    cursor.execute("PRAGMA cache_size = -32000;")
+    cursor.execute("PRAGMA busy_timeout = 10000;")
+    cursor.close()
```
Comment (mikeldking) [inline src/phoenix/db/database.py:92]: oh interesting. Does this need to happen every time or can it be done on the first migration?

File: src/phoenix/db/database.py [hunk 4]
```diff
@@ -78,6 +108,18 @@ def __init__(self, database: Optional[Path] = None) -> None:
...
 
+        engine = (
+            create_engine(f"sqlite:///{db_path}", echo=True)
+            if db_path
+            else create_engine(
+                "sqlite:///:memory:",
+                echo=True,
+                creator=_mem_db_creator,
+            )
+        )
```
Comment (mikeldking) [inline src/phoenix/db/database.py:119]: I think we are gonna want this to be declarative

File: src/phoenix/db/database.py [hunk 5]
```diff
@@ -51,24 +62,43 @@
...
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA foreign_keys = ON;")
+    cursor.execute("PRAGMA journal_mode = WAL;")
+    cursor.execute("PRAGMA synchronous = OFF;")
+    cursor.execute("PRAGMA cache_size = -32000;")
+    cursor.execute("PRAGMA busy_timeout = 10000;")
+    cursor.close()
+
+
 class SqliteDatabase:
```
Comment (mikeldking) [inline src/phoenix/db/database.py:95]: I'm not sure this singleton makes sense long term - just making sure we are on the same page here - there's gonna need to be different query patterns I think and trying to do it all via one DB object will be difficult.

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/db/database.py [hunk 6]
```diff
@@ -3,12 +3,23 @@
...
+from phoenix.trace.schemas import (
+    ComputedValues,
+    Span,
+    SpanContext,
+    SpanEvent,
+    SpanKind,
+    SpanStatusCode,
+)
 
 _CONFIG = """
```
[the remark this answers is not in the document]
Comment (RogerHYang) [reply src/phoenix/db/database.py:24]: yes, just keeping these here for now so the code still runs, until i switch the rest over to the ORM

File: src/phoenix/db/database.py (hunk 3 above)
Comment (RogerHYang) [reply src/phoenix/db/database.py:92]: The first two are persisted (i think). But the rest are run-time configs.

File: src/phoenix/db/database.py (hunk 5 above)
Comment (RogerHYang) [reply src/phoenix/db/database.py:95]: Yea, agree

File: src/phoenix/db/database.py (hunk 4 above)
Comment (RogerHYang) [reply src/phoenix/db/database.py:119]: yea, agree

**Pull request merged**